components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/internal/util/AccessTokenEventUtil.java

```diff
@@ -221,4 +230,71 @@ private static void publish(Map<String, Object> properties) {
             LOG.warn("Error occurred publishing event " + IdentityEventConstants.Event.TOKEN_REVOKED, e);
         }
     }
+
+    /**
+     * Publishes an event when a token is issued.
+     *
+     * @param tokReqMsgCtx            The token request message context containing information about the token request.
+     * @param oAuth2AccessTokenReqDTO The OAuth2 access token request DTO containing details about the access token
+     *                                request.
+     * @throws UserIdNotFoundException If the user ID cannot be found in the context.
+     */
+    public static void publishTokenIssueEvent(OAuthTokenReqMessageContext tokReqMsgCtx,
+                                              OAuth2AccessTokenReqDTO oAuth2AccessTokenReqDTO)
+            throws UserIdNotFoundException {
+
+        HashMap<String, Object> properties = new HashMap<>();
+
+        OauthTokenIssuer tokenIssuer = null;
+        try {
+            tokenIssuer = OAuth2Util.getOAuthTokenIssuerForOAuthApp(oAuth2AccessTokenReqDTO.getClientId());
+        } catch (IdentityOAuth2Exception e) {
+            LOG.error("Error while retrieving the OAuth token issuer for client ID: " +
+                    oAuth2AccessTokenReqDTO.getClientId(), e);
+        } catch (InvalidOAuthClientException e) {
+            LOG.error("Invalid OAuth client with client ID: " + oAuth2AccessTokenReqDTO.getClientId(), e);
+        }
+        if (tokenIssuer != null) {
+            properties.put(IdentityEventConstants.EventProperty.TOKEN_TYPE, tokenIssuer.getAccessTokenType());
+        }
+
+        if (tokReqMsgCtx != null) {
+            if (tokReqMsgCtx.getAuthorizedUser() != null) {
+                properties.put(IdentityEventConstants.EventProperty.USER_ID,
+                        tokReqMsgCtx.getAuthorizedUser().getUserId());
+                properties.put(IdentityEventConstants.EventProperty.USER_NAME,
+                        tokReqMsgCtx.getAuthorizedUser().getUserName());
+                properties.put(IdentityEventConstants.EventProperty.USER_STORE_DOMAIN,
+                        tokReqMsgCtx.getAuthorizedUser().getUserStoreDomain());
+                properties.put("IS_ORGANIZATION_USER", tokReqMsgCtx.getAuthorizedUser().isOrganizationUser());
```

Comment:
Shall we replace these with the framework constants

Reply:
Fixed 7802dae7753b368615c4d3d8683cfbb12f3f91e4